tests/test_allow_none_vs_default.py

```diff
@@ -0,0 +1,191 @@
+"""Copyright 2019 Equinor ASA and The Netherlands Organisation for
+Applied Scientific Research TNO.
+
+Licensed under the MIT license.
+
+Permission is hereby granted, free of charge, to any person obtaining a
+copy of this software and associated documentation files (the
+"Software"), to deal in the Software without restriction, including
+without limitation the rights to use, copy, modify, merge, publish,
+distribute, sublicense, and/or sell copies of the Software, and to
+permit persons to whom the Software is furnished to do so, subject to
+the conditions stated in the LICENSE file in the project root for
+details.
+
+The above copyright notice and this permission notice shall be included
+in all copies or substantial portions of the Software.
+"""
+
+
+import unittest
+
+
+import configsuite
+from configsuite import MetaKeys as MK
+from configsuite import types
+
+
+class TestNotAllowNoneVsDefault(unittest.TestCase):
+    def test_allow_none_default_not_required(self):
+        schema = {
+            MK.Type: types.NamedDict,
+            MK.Content: {
+                "my_value": {
+                    MK.Type: types.Integer,
+                    MK.AllowNone: True,
+                    MK.Required: False,
+                    MK.Default: 0,
+                },
+            },
+        }
+
+        for value in (-1, 4, 1000, None):
+            suite = configsuite.ConfigSuite({"my_value": value}, schema)
+            self.assertTrue(suite.valid)
+            self.assertEqual(value, suite.snapshot.my_value)
+
+        suite = configsuite.ConfigSuite({}, schema)
+        self.assertTrue(suite.valid)
+        self.assertEqual(0, suite.snapshot.my_value)
+
+    def test_allow_none_no_default_not_required(self):
+        schema = {
+            MK.Type: types.NamedDict,
+            MK.Content: {
+                "my_value": {
+                    MK.Type: types.Integer,
+                    MK.AllowNone: True,
+                    MK.Required: False,
+                },
+            },
+        }
+
+        for value in (-1, 4, 1000, None):
+            suite = configsuite.ConfigSuite({"my_value": value}, schema)
+            self.assertTrue(suite.valid)
+            self.assertEqual(value, suite.snapshot.my_value)
+
+        suite = configsuite.ConfigSuite({}, schema)
+        self.assertTrue(suite.valid)
+        self.assertEqual(None, suite.snapshot.my_value)
+
+    def test_disallow_none_default_not_required(self):
```

Comment:
Isn't this just testing that `MK.AllowNone` is implemented properly?

If `MK.AllowNone` was not permitted to be set to `False` in this case, it shouldn't be a `valid` configsuite instance.